### PR TITLE
Add CMake build files

### PR DESCRIPTION
**Pull request by neugepower**
This PR adds all CMake build files to support a full end-to-end build with CMake.

The following things have been tested:
- [x] `Unix Makefile` and `Ninja` generator
- [x] Unit test compilation and execution
- [x] On-the-fly implementation generation for serializable classes
- [x] QtCreator build
- [x] Linux build via `Dockerfile.build` (part of this PR)

The PR doesn't include:
- CMake installation logic to install libraries and headers

Since there are only files added and no files changed this is a fully backward-compatible
change which should continue to support the old way of building via QtCreator.

Please note that the CMake build relies on all libraries installed on the system. See `Dockerfile.build` for necessary dependencies.

Please also note that there is a CMake module that should allow upstream projects to generate implementations easily.

Last but not least, this hasn't been tested yet against a downstream project. In this version all the interfaces are private.
I suspect some additional changes that are needed to easily integrate this CMake build into a downstream project. This can
be either done via this PR or an additional future PR.

### CODE REVIEW

File: cflib/db/psql.cpp [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * Licensed under the MIT License.
  */
 
-#include "psql.h"
+#include <cflib/db/psql.h>
```
Comment (neugepower) [inline cflib/db/psql.cpp:8]: Drive-by change! Please check.

**Comment by neugepower** [comment]
Draft: the public interface is not complete yet. Want to do that first so that projects can consume `cflib` with headers.

File: cflib/db/psql.cpp (hunk 1 above)
Comment (neugepower) [reply cflib/db/psql.cpp]: Removed. This seems to be common practice in `cflib`.

**Review by fishbach — APPROVED** [review]
